internal/config/main.go

```diff
@@ -52,6 +54,7 @@ var (
 		TLSInsecure:   flag.Bool("insecure", false, "Allow insecure certificates"),
 		TlsExtraCAs:   flag.String("extra-ca-certs", "", "comma-separated files containing extra PEMs to trust for TLS connections in addition to the system trust store"),
 		MaxBGPPaths:   flag.Int("max-bgp-paths", 10000, "How many BGP Paths to receive when counting routes, needs to be higher then the number of routes or metrics will not be generated"),
+		MaxVPNUsers:   flag.Int("max-vpn-users", 10000, "How many VPN Users to receive when counting users, needs to be greater than or equal the number of users or metrics will not be generated"),
```

Comment:
This makes a few warning bells go off in my head. I am thinking two issues:

1. The sheer amount of metrics this will create is likely quite huge given it is per user. It is configurable for the maximum number of users though, so that's OK.
2. If the iteration order changes, so does the potential order of the output - which is fine, given that we emit no partial metrics (this would be a problem if we e.g. emitted *up to* 1000 first users and capped it at that).

I think this will work, and I can see why people would like to have this metric available. It scares me though.

Could we lower the default limit to 100 users maybe? For the BGP case we do not (unless I remember incorrectly?) emit a metric for every path we fetch. Here the very likely case is that every user will have a small amount of sessions, so I think limiting it to a maximum of 100 metrics would be a sensible default.

Reply:
>     1. The sheer amount of metrics this will create is likely quite huge given it is per user. It is configurable for the maximum number of users though, so that's OK.

This can for sure create a lot of metrics. Cardinality is always an issue for this kind of metrics.

Reply:
>     2. If the iteration order changes, so does the potential order of the output - which is fine, given that we emit no partial metrics (this would be a problem if we e.g. emitted _up to_ 1000 first users and capped it at that).

Yes, there should be no partial metrics; therefore we can expect consistency across scrapes.

Reply:
> I think this will work, and I can see why people would like to have this metric available. It scares me though.

My original aim is to be able to trigger an alert if a user is connected with SSL VPN in certain conditions.
For example:
-	User is connect while he is not supposed to. This could be acting as a double check for highly critical access where fortigate configuration is already pretty restricted.
-	User is connect at some hours it should not. IT team member of the company for example have emergency access but should use it only if required.

For our needs, this will only be used for detecting abnormal security behavior.
No aim to spy on users activity, but I am not responsible for people motivations outside of security.
We can just advise people not to be evil.

Reply:
> Could we lower the default limit to 100 users maybe? For the BGP case we do not (unless I remember incorrectly?) emit a metric for every path we fetch. Here the very likely case is that every user will have a small amount of sessions, so I think limiting it to a maximum of 100 metrics would be a sensible default.

Ok for a default limit of 100. I guess this fit most deployment scenarios.
To force people not to be evil we can also lower it to 0 by default and add a check to not run the API call if set to 0.
Therefore, only users that want to retrieve this metric would have to set the parameter to a higher value they will have to choose.
What do you think of disabling this metric by default by setting the limit to 0 ?

Reply:
@bluecmd @secustor Do you think disabling this probe by default by setting it to 0 is fine? Let me know and I will update code accordingly

Reply:
I am OK with setting the value to 0 or a small value (<100) by default. Either way works for me.

Can you rebase (not merge) on the latest? It seems there are conflicts and the diffs are all messed up sadly.

Reply:
Sorry for the mess. I am not very confident with rebase, I never used it before. Hope this is better now. If not tell me, and I will try harder.

Reply:
It's fine, it's one of the harder things to do in git.

You want to do:
```
$ git fetch origin
$ git rebase -i origin/master
$ git push --force-with-lease
```

When you do `git rebase -i` you will see the commits that it wants to bring over "ontop" of the new `master`. Those commits should be the ones that contain your modifications, nothing else.